### PR TITLE
SUS-767: Show "What links here" and other useful tools on file pages

### PR DESCRIPTION
**Pull request by mszabo-wikia**
ticket & QA notes: https://wikia-inc.atlassian.net/browse/SUS-767

@mixth-sense

### CODE REVIEW

**Comment by mszabo-wikia** [comment]
Failing test is unrelated to this commit

File: includes/wikia/services/usercommands/UserCommand.php [hunk 1]
```diff
@@ -1,214 +1,357 @@
...
-			$this->needData();
-			return $this->description;
-		}
+	/**
+	 * Returns the attributes used for the <a> element when outputting this tool
+	 * @return array key-value array of attributes and their values
+	 */
+	protected function getLinkAttributes() {
+		$attributes = [];
+		$attributes['data-tool-id'] = $this->id;
```
Comment (vforge) [inline includes/wikia/services/usercommands/UserCommand.php]: You can put those values into initialization of an array above.

File: includes/wikia/services/usercommands/UserCommand.php [hunk 2]
```diff
@@ -1,214 +1,357 @@
...
+		$attributes['data-name'] = $this->getTrackerName();
+		if ( $this->href ) $attributes['href'] = $this->href;
```
Comment (vforge) [inline includes/wikia/services/usercommands/UserCommand.php]: Please do not use inline `if` statements.

File: includes/wikia/services/usercommands/UserCommand.php [hunk 3]
```diff
@@ -1,214 +1,357 @@
...
-			if (!$this->available) {
-				return '';
-			}
-
-			$html = '';
-			$html .= Xml::openElement('li',$this->getListItemAttributes());
-
-			$html .= $this->renderIcon();
+		$html = '';
+		$html .= Xml::openElement( 'li', $this->getListItemAttributes() );
```
Comment (vforge) [inline includes/wikia/services/usercommands/UserCommand.php]: Those two lines can be simplified.

File: includes/wikia/services/usercommands/UserCommand.php [hunk 4]
```diff
@@ -1,214 +1,357 @@
...
+	 * Returns information for UserToolsController Menu template to output this tool
+	 * @return array|bool Array of information used by UserToolsController Menu template to output this tool, or false if disabled
+	 */
+	public function getRenderData() {
+		$this->needData();
+		if ( !$this->available )
+			return false;
+
+		$data = [
+			'type'         => $this->defaultRenderType,
```
Comment (vforge) [inline includes/wikia/services/usercommands/UserCommand.php]: Don't use whitespaces for formatting.

File: includes/wikia/services/usercommands/UserCommand.php [hunk 5]
```diff
@@ -1,214 +1,357 @@
...
+	}
 
-			return $data;
+	/**
+	 * Returns information for UserToolsController Menu template to output this tool
+	 * @return array|bool Array of information used by UserToolsController Menu template to output this tool, or false if disabled
+	 */
+	public function getRenderData() {
+		$this->needData();
+		if ( !$this->available )
```
Comment (vforge) [inline includes/wikia/services/usercommands/UserCommand.php]: Don't use inline `if`s, especially that way :)

Please convert them (and all the rest of them) to blocks.

**Comment by mszabo-wikia** [comment]
@vforge Done, thanks for the review! :)

**Comment by vforge** [comment]
LGTM

**Pull request merged**